commit 08f5083d541ee47ac04b5bcff6756bbf1c81da07
Author: Heiner Kallweit <hkallweit1@gmail.com>
Date:   Sun Aug 12 13:26:26 2018 +0200

    r8169: don't use MSI-X on RTL8168g
    
    [ Upstream commit 7c53a722459c1d6ffb0f5b2058c06ca8980b8600 ]
    
    There have been two reports that network doesn't come back on resume
    from suspend when using MSI-X. Both cases affect the same chip version
    (RTL8168g - version 40), on different systems. Falling back to MSI
    fixes the issue.
    Even though we don't really have a proof yet that the network chip
    version is to blame, let's disable MSI-X for this version.
    
    Reported-by: Steve Dodd <steved424@gmail.com>
    Reported-by: Lou Reed <gogen@disroot.org>
    Tested-by: Steve Dodd <steved424@gmail.com>
    Tested-by: Lou Reed <gogen@disroot.org>
    Fixes: 6c6aa15fdea5 ("r8169: improve interrupt handling")
    Signed-off-by: Heiner Kallweit <hkallweit1@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/realtek/r8169.c b/drivers/net/ethernet/realtek/r8169.c
index 764b25fa470c..d19065857afe 100644
--- a/drivers/net/ethernet/realtek/r8169.c
+++ b/drivers/net/ethernet/realtek/r8169.c
@@ -8066,6 +8066,11 @@ static int rtl_alloc_irq(struct rtl8169_private *tp)
 		RTL_W8(tp, Config2, RTL_R8(tp, Config2) & ~MSIEnable);
 		RTL_W8(tp, Cfg9346, Cfg9346_Lock);
 		flags = PCI_IRQ_LEGACY;
+	} else if (tp->mac_version == RTL_GIGA_MAC_VER_40) {
+		/* This version was reported to have issues with resume
+		 * from suspend when using MSI-X
+		 */
+		flags = PCI_IRQ_LEGACY | PCI_IRQ_MSI;
 	} else {
 		flags = PCI_IRQ_ALL_TYPES;
 	}
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1014_08f5083d541ee47ac04b5bcff6756bbf1c81da07.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1014_08f5083d541ee47ac04b5bcff6756bbf1c81da07.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
